test/FunctionsV2/DurableHttpTests.cs

```diff
@@ -0,0 +1,958 @@
+                    httpMethod: HttpMethod.Get,
+                    headers: headersTwo);
+
+                string functionName = nameof(TestOrchestrations.CallHttpAsyncOrchestrator);
+                var clientTwo = await host.StartOrchestratorAsync(functionName, testRequestTwo, this.output);
+                var statusTwo = await clientTwo.WaitForCompletionAsync(this.output, timeout: TimeSpan.FromSeconds(Debugger.IsAttached ? 3000 : 90));
+                var outputTwo = statusOne?.Output;
+                DurableHttpResponse responseTwo = outputOne.ToObject<DurableHttpResponse>();
+
+                Assert.Equal(HttpStatusCode.OK, responseTwo.StatusCode);
+
+                await host.StopAsync();
+            }
+        }
+
+        /// <summary>
+        /// End-to-end test which checks if the CallHttpAsync Orchestrator returns an OK (200) status code
+        /// when a Bearer Token is added to the DurableHttpRequest object.
+        /// </summary>
+        [Theory]
+        [Trait("Category", PlatformSpecificHelpers.TestCategory)]
+        [MemberData(nameof(TestDataGenerator.GetFullFeaturedStorageProviderOptions), MemberType = typeof(TestDataGenerator))]
+        public async Task DurableHttpAsync_Synchronous_AddsBearerToken(string storageProvider)
+        {
+            HttpMessageHandler httpMessageHandler = CreateSynchronousHttpMessageHandlerForTestingTokenSource();
+
+            using (JobHost host = TestHelpers.GetJobHost(
+                this.loggerProvider,
+                nameof(this.DurableHttpAsync_AsynchronousAPI_ReturnsOK200),
+                enableExtendedSessions: false,
+                storageProviderType: storageProvider,
+                durableHttpMessageHandler: new DurableHttpMessageHandlerFactory(httpMessageHandler)))
+            {
+                await host.StartAsync();
+
+                Dictionary<string, string> headers = new Dictionary<string, string>();
+                headers.Add("Accept", "application/json");
+                MockTokenSource mockTokenSource = new MockTokenSource("dummy test token");
+
+                TestDurableHttpRequest testRequest = new TestDurableHttpRequest(
+                    httpMethod: HttpMethod.Get,
+                    headers: headers,
+                    tokenSource: mockTokenSource);
+
+                var client = await host.StartOrchestratorAsync(nameof(TestOrchestrations.CallHttpAsyncOrchestrator), testRequest, this.output);
+                var status = await client.WaitForCompletionAsync(this.output, timeout: TimeSpan.FromSeconds(Debugger.IsAttached ? 3000 : 90));
+                var output = status?.Output;
+                DurableHttpResponse response = output.ToObject<DurableHttpResponse>();
+
+                Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+
+                await host.StopAsync();
+            }
+        }
+
+        private static HttpMessageHandler CreateSynchronousHttpMessageHandlerForTestingTokenSource()
```

Comment:
NIT: I would replace the word `Create` with `Mock`, so it is clear you are mocking this behavior as to using an actual HttpMessageHandler.